pkg/rslog/rslogtest/logger_impl_test.go

```diff
@@ -352,3 +354,101 @@ func (s *LoggerImplTestSuite) TestNewDiscardingLogger() {
 
 	s.NotNil(discardingLogger)
 }
+
+func (s *LoggerImplTestSuite) TestUTCJSONFormatter() {
+	// create a test location 1 hr east of UTC
+	loc := time.FixedZone("test/zone", 3600)
+	utc := time.Now().UTC()
+
+	// shift UTC time to the test location
+	nonUtc := utc.In(loc)
+
+	// verify shifted time is 1 hr ahead
+	expected := utc.Add(time.Hour)
+
+	expYear, expMonth, expDay := expected.Date()
+	nonUtcYear, nonUtcMonth, nonUtcDay := nonUtc.Date()
+
+	s.Equal(expYear, nonUtcYear)
+	s.Equal(expMonth, nonUtcMonth)
+	s.Equal(expDay, nonUtcDay)
+
+	expHour, expMin, expSec := expected.Clock()
+	nonUtcHour, nonUtcMin, nonUtcSec := nonUtc.Clock()
+
+	s.Equal(expHour, nonUtcHour)
+	s.Equal(expMin, nonUtcMin)
+	s.Equal(expSec, nonUtcSec)
+
+	// verify shifted time is not in UTC location
+	s.NotEqual(utc.Location(), nonUtc.Location())
+
+	formatter := rslog.NewUTCJSONFormatter()
+
+	// create a log entry with non-UTC time
+	entry := &logrus.Entry{
+		Level:   logrus.DebugLevel,
+		Time:    nonUtc,
+		Message: "test",
+	}
+
+	serialized, err := formatter.Format(entry)
+
+	s.Nil(err)
+
+	result := string(serialized)
+
+	// assert that UTC time appears in the formatted entry
+	utcTimestamp := utc.Format(formatter.TimestampFormat)
+	s.True(strings.Contains(result, utcTimestamp))
+
+}
+
+func (s *LoggerImplTestSuite) TestUTCTextFormatter() {
+
+	// create a test location 1 hr east of UTC
+	loc := time.FixedZone("test/zone", 3600)
+	utc := time.Now().UTC()
+
+	// shift UTC time to the test location
+	nonUtc := utc.In(loc)
+
+	// verify shifted time is 1 hr ahead
+	expected := utc.Add(time.Hour)
+
+	expYear, expMonth, expDay := expected.Date()
+	nonUtcYear, nonUtcMonth, nonUtcDay := nonUtc.Date()
+
+	s.Equal(expYear, nonUtcYear)
+	s.Equal(expMonth, nonUtcMonth)
+	s.Equal(expDay, nonUtcDay)
+
+	expHour, expMin, expSec := expected.Clock()
+	nonUtcHour, nonUtcMin, nonUtcSec := nonUtc.Clock()
+
+	s.Equal(expHour, nonUtcHour)
+	s.Equal(expMin, nonUtcMin)
+	s.Equal(expSec, nonUtcSec)
+
+	// verify shifted time is not in UTC location
+	s.NotEqual(utc.Location(), nonUtc.Location())
```

Comment:
You might consider moving lines `359-384` and `409-434` to a helper method and invoking it here like:

```
utc, nonUtc := getTestTimes(c)
```

That would help to clarify exactly what we're testing in these two unit tests. I was about to comment that we're testing the `time` package in these lines, but then I realized that it is useful to prove that we've actually created the times we intended.

Reply:
good call - just refactored those lines. 

Does my change look ok in terms of go best practices?

Reply:
Looks great to me! Thanks.